src/transformers/integrations/peft.py

```diff
@@ -129,12 +133,27 @@ def load_adapter(
             adapter_state_dict (`Dict[str, torch.Tensor]`, *optional*):
                 The state dict of the adapter to load. This argument is used in case users directly pass PEFT state
                 dicts
+            low_cpu_mem_usage (`bool`, *optional*, defaults to `False`):
+                Reduce memory usage while loading the PEFT adapter. This should also speed up the loading process.
+                Requires PEFT version 0.13.0 or higher.
             adapter_kwargs (`Dict[str, Any]`, *optional*):
                 Additional keyword arguments passed along to the `from_pretrained` method of the adapter config and
                 `find_adapter_config_file` method.
         """
         check_peft_version(min_version=MIN_PEFT_VERSION)
 
+        # peft only supports low_cpu_mem_usage starting from v0.13.0
+        peft_load_kwargs = {}
```

Comment:
LGTM, not sure we need all the kwargs as you are not taking them from the kwargs of this function 😉

Reply:
Not sure if I get you, do you mean this should be merged with `adapter_kwargs`? That wouldn't work, as these are `kwargs` used for a different purpose. 

Or that using a dict for a single argument is overkill?

Reply:
> Or that using a dict for a single argument is overkill?
this 🤗

Reply:
I found this more readable and extensible than having:

```python
if low_cpu_mem_usage:
    inject_adapter_in_model(peft_config, self, adapter_name, low_cpu_mem_usage=low_cpu_mem_usage)
else:
    inject_adapter_in_model(peft_config, self, adapter_name)
```

(same with the `set_peft_model_state_dict` call)

If you want me to change it to this instead or have an alternative idea, let me know and I'll change it.

Reply:
no worries